modules/importer/src/server/csaf/mod.rs

```diff
@@ -38,13 +38,24 @@ impl super::Server {
         let fetcher = Fetcher::new(Default::default()).await?;
         let options = HttpOptions::new().since(last_success);
 
-        let source = match Url::parse(&importer.source) {
-            Ok(url) => HttpSource::new(url, fetcher, options),
+        let source: DispatchSource = match Url::parse(&importer.source) {
```

Comment:
Should we really consider adding this, we should leverage the existing `new_source` function, which already does all the handling required:

https://github.com/ctron/csaf-walker/blob/f018d47cc5025ddf3f3a855f942cdf1f72726b3b/csaf/src/source/mod.rs#L41-L49